src/components/JourneyDisplay.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { GiSnail } from 'react-icons/gi';
+import { HiArrowCircleRight, HiClock } from 'react-icons/hi';
+import { SiPuma } from 'react-icons/si';
+import StationDisplay from './JourneyDisplayComponents/StationDisplay';
+import StatusDisplay from './JourneyDisplayComponents/StatusDisplay';
+import TimeDisplay from './JourneyDisplayComponents/TimeDisplay';
+import { JourneyInfo } from './MainPage';
+
+interface JourneyDisplayArgs {
+    journeyInfo: JourneyInfo;
+}
+
+const JourneyDisplay: React.FC<JourneyDisplayArgs> = ({ journeyInfo }) => {
+
+    const minutesToHoursAndMinutes = (minutes : number): string => {
+        if (minutes < 60) {
+            return minutes + ' min';
+        } else {
+            const hours  = Math.floor(minutes / 60);
+            const remainingMinutes = minutes % 60;
+            return hours + 'h ' + remainingMinutes + 'min';
+        }
+    };
+
+    return (
+        <div className = "journey-display" 
+            style = { { backgroundColor: journeyInfo.isFastestJourney ? '#bde8ae' : journeyInfo.isOvertaken ? '#e2db6f' : 'white' } }>
```

Comment:
This might be neater if the background colour was due to the class name, and the class name changes depending on isFastestJourney?
The icon could also depend on the class name so you would not need to do all these if statements below as well

Reply:
used class names for this